test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,200 @@
+ */
+
+public class JTableRightAlignmentTest {
+    static JFrame frame;
+    static CustomTable table;
+    static boolean passed = true;
+    static String failureString;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        Robot robot = new Robot();
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Test JTable");
+
+                JPanel panel = new JPanel(new GridBagLayout());
+                frame.setContentPane(panel);
+                table = new CustomTable();
+                panel.add(new JScrollPane(table.table),
+                        new GridBagConstraints(0, 0, -1, -1, 1.0, 1.0,
+                                GridBagConstraints.PAGE_START, GridBagConstraints.BOTH,
+                                new Insets(2, 2, 2, 2), 0, 0));
+                frame.pack();
+                frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+                frame.applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+                frame.setVisible(true);
+            });
+
+            robot.waitForIdle();
+            robot.delay(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                int maxHeight = (int) (((double) table.table.getTableHeader().getHeight())
+                        + ((double) table.table.getHeight()));
+                int yPos = table.table.getTableHeader().getLocationOnScreen().y;
+                int xPos = table.table.getLocationOnScreen().x + table.table.getWidth() -
+                        table.table.getTableHeader().getColumnModel()
+                                .getColumn(2)
+                                .getWidth() - 1;
+                Color expectedRGB = robot.getPixelColor(xPos, yPos);
+                for (int y = yPos; y < yPos + maxHeight; y++) {
+                    if (expectedRGB.getRGB() != robot.getPixelColor(xPos, y).getRGB()) {
+                        BufferedImage failImage = robot.createScreenCapture(
+                                new Rectangle(xPos, yPos, 3, maxHeight));
+                        saveImage(failImage, "failureImage.png");
+                        passed = false;
+                        failureString = "Test Failed at <" + xPos + ", " + y + ">";
+                        break;
+                    }
+                }
+            });
+            if (!passed) {
+                throw new RuntimeException(failureString);
+            } else {
+                System.out.println("Test Passed!");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
```

Comment:
Is it OK to save an image here? In the past, I stored my images similar to in #7310 to store the image in the correct area for temporary images for a test.

Reply:
U mean to create a test directory and store image there?

Reply:
When run with jtreg, the current directory is the scratch directory which is thrown away if the test passes or stored by the harness if the test fails. It is safe to use it this way, many tests do save images like this.

Your case was somewhat different: you creating and saving an image for the test use. I guess it could also do without the extra stuff, however, preserving the generated image if the test fails may be unnecessary.